pkg/apis/serving/v1alpha2/kfservice_types.go

```diff
@@ -21,36 +21,65 @@ import (
 
 // KFServiceSpec defines the desired state of KFService
 type KFServiceSpec struct {
-	Default ModelSpec `json:"default"`
+	Default ComponentsSpec `json:"default"`
 	// Canary defines an alternate configuration to route a percentage of traffic.
-	Canary               *ModelSpec `json:"canary,omitempty"`
-	CanaryTrafficPercent int        `json:"canaryTrafficPercent,omitempty"`
+	Canary               *ComponentsSpec `json:"canary,omitempty"`
+	CanaryTrafficPercent int             `json:"canaryTrafficPercent,omitempty"`
 }
 
-// ModelSpec defines the configuration to route traffic to a predictor.
-type ModelSpec struct {
+type ComponentsSpec struct {
```

Comment:
Use singular instead? `ComponentSpec`

Reply:
What about something like EndpointSpec or RouteSpec?

Canary and Default are effectively different "endpoints". This maps somewhat cleanly to AWS's data model of Models and Endpoints. We've just cleaned it up a bit :).

Reply:
Isn't it a collection of 'components' or 'endpoints'.....? Hence plural? Also we are switching between noun and verb for different spec names.... should we be consistent?

Reply:
If everyone agrees I am ok to change to noun :)